app/src/main/java/lt/vilnius/tvarkau/SignInActivity.java

```diff
@@ -0,0 +1,104 @@
+package lt.vilnius.tvarkau;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+
+/*
+    Activity which handles log in of user.
+    It does not matter whether user decides to do it,
+    he will be transferred to MainActivity
+
+ */
+public class SignInActivity extends AppCompatActivity implements GoogleApiClient.OnConnectionFailedListener{
```

Comment:
I propose to remove the login button from home screen and show it up in the following cases:

1. The user doesn't have a profile yet and clicks My Profile on home screen for the first time
2. The user doesn't have a profile yet and selects "Contact me in case of need" in Nauja problema screen.

![my_profile](https://cloud.githubusercontent.com/assets/15831041/12529630/27fcf0fc-c1c7-11e5-9f5f-6c2dbd76fa0e.png)